scripts/update_package_versions.py

```diff
@@ -712,54 +588,298 @@ def bump_version_in_yaml(
+                    print(f"Version parse error: {e}. Enter a new one")
+                    continue
+        elif self.option_update_version == "minor":
+            new_version = ver.bump_minor()
+        elif self.option_update_version == "patch":
+            new_version = ver.bump_patch()
+        else:
+            raise Exception("unknown version update mode")
+
+        return str(new_version)
+
+    def bump_package_version(
+        self,
+        current_public_id: PublicId,
+        configuration_file_path: Path,
+        type_: str,
+        is_ambiguous: bool = False,
+    ) -> None:
+        """
+        Bump the version references of the package in the repo.
+
+        Includes, bumping the package itself.
+
+        :param current_public_id: the current public id
+        :param configuration_file_path: the path to the configuration file
+        :param type_: the type of package
+        :param is_ambiguous: whether or not the package id is ambiguous
+        """
+        new_version = self.get_new_package_version(current_public_id)
+
+        new_public_id = PublicId(
+            current_public_id.author, current_public_id.name, new_version
+        )
+        for rootdir in DIRECTORIES:
+            for path in Path(rootdir).glob("**/*"):
+                if path.is_file() and str(path).endswith(
+                    (".py", ".yaml", ".md", ".sh")
+                ):
+                    self.inplace_change(
+                        path, current_public_id, new_public_id, type_, is_ambiguous,
+                    )
+
+        bump_version_in_yaml(configuration_file_path, type_, new_public_id.version)
+
+    def _run_hashing(self):
+        while self._run_once():
+            self._run_hashing()
+
+    def inplace_change(
+        self,
+        fp: Path,
+        old_public_id: PublicId,
+        new_public_id: PublicId,
+        type_: str,
+        is_ambiguous: bool,
+    ) -> None:
+        """Replace the occurrence of a string with a new one in the provided file."""
+
+        content = fp.read_text()
+        if not file_should_be_processed(content, old_public_id):
+            return
+
+        old_string = str(old_public_id)
+        new_string = str(new_public_id)
+        print(
+            f"Processing file {fp} for replacing {old_string} with {new_string} (is_ambiguous: {is_ambiguous})"
+        )
+
+        content = replace_in_yamls(content, old_public_id, new_public_id, type_)
+        content = replace_in_protocol_readme(
+            fp, content, old_public_id, new_public_id, type_
+        )
+        if not is_ambiguous:
+            content = content.replace(old_string, new_string)
+        else:
+            content = self._ask_to_user_and_replace_if_allowed(
+                content, old_string, new_string, type_
+            )
+
+        with fp.open(mode="w") as f:
+            f.write(content)
+
+    def _ask_to_user_and_replace_if_allowed(
```

Comment:
Same as below. Better to rename `_ask_to_user_and_replace_if_allowed` with `_ask_user_and_replace_if_allowed`

Reply:
updated